app/decorators/proposal_decorator.rb

```diff
@@ -9,6 +9,38 @@ def total_approvers
     object.individual_steps.count
   end
 
+  def final_completed_date
+    if completed? && total_approvers > 0
+      object.individual_steps.last.completed_at
+    else
+      ""
+    end
+  end
+
+  def total_completion_days
+    if completed?
+      (final_completed_date.to_date - created_at.to_date).to_i
+    else
+      ""
+    end
+  end
+
+  def final_step_label
+    if total_approvers > 0
+      "Final #{object.individual_steps.last.decorate.label} Completed"
+    else
+      "Final Step Completed"
+    end
+  end
+
+  def self.final_step_label(proposal = nil)
```

Comment:
in that case maybe it makes sense to have a helper method to handle this? I think I understand the need to have something other than an instance method, but I find the class method confusing because usually a class method is called on a collection rather than a single object or no objects.